MobileCenterAnalytics/MobileCenterAnalyticsTests/MSSessionTrackerTests.m

```diff
@@ -196,4 +196,38 @@ - (void)testStartSessionOnAppForegrounded {
                             withPriority:MSPriorityDefault]);
 }
 
+- (void)testOnProcessingLogWithPriority {
+
+  // When
+  MSLogWithProperties *log = [MSLogWithProperties new];
+
+  // Then
+  XCTAssertNil(log.sid);
+  XCTAssertNil(log.toffset);
+
+  [self.sut onProcessingLog:log withPriority:MSPriorityDefault];
+
+  // And then
+  XCTAssertNil(log.toffset);
+  XCTAssertEqual(log.sid, self.sut.sessionId);
+
+  // When
+  log.toffset = 0;
+
+  [self.sut onProcessingLog:log withPriority:MSPriorityDefault];
+
+  // Then
+  XCTAssertEqual(0, log.toffset.integerValue);
+  XCTAssertEqual(log.sid, [self.sut.pastSessions firstObject].sessionId);
+
+  // When
+  log.toffset = [NSNumber numberWithUnsignedLongLong:UINT64_MAX];
+
```

Comment:
No, we can't, because we should check what we get last session id if we have the biggest time offset.